CorsixTH/Lua/app.lua

```diff
@@ -744,8 +744,18 @@ function App:_loadLevel(level, difficulty, level_name, level_file, level_intro,
   self.map:setBlocks(self.gfx:loadSpriteTable("Data", "VBlk-0"))
   self.map:setDebugFont(self.gfx:loadFont("QData", "Font01V"))
 
+  -- In Free Build mode?
+  local free_build_mode
+  local isMainCampaign = tonumber(new_map.level_number)
+  local isCustomCampaign = campaign_info ~= nil
+  if isMainCampaign or isCustomCampaign then
+    free_build_mode = false -- disable free_build_mode for any Campaign
+  else
+    free_build_mode = self.config.free_build_mode
+  end
```

Comment:
> This could fit nicer in a local function.

I agree. 

> NB: locals almost always use lower snake case; except constants (and then then sometimes we still don't lol).

Oh, true, my miss.

Reply:
> This could fit nicer in a local function.

Thanks. Applied.